[Magic Firewall] Full packet captures (#5365)
cloudflare/cloudflare-docs: content/magic-firewall/how-to/collect-pcaps.md

---
title: Collect PCAPs
pcx-content-type: how-to
weight: 5
---

# Collect packet captures (PCAPs)

> After a packet capture is requested and the capture is collected, the output is contained within one or more files in PCAP  file format. Before starting a `full` type packet capture, you must first follow instructions for [configuring a bucket](../pcaps-bucket-setup).

Reviewer: ```suggestion
After a packet capture is requested and the capture is collected, the output is contained within one or more files in PCAP file format. Before starting a `full` type packet capture, you must first follow instructions for [configuring a bucket](../pcaps-bucket-setup).
```

{{<Aside>}}

This feature is available for Advanced Magic Firewall users. For access, contact your account team.

{{</Aside>}}

## Send a packet capture request

Currently, when a packet capture is requested, packets flowing at the edge through the Magic Transit system are captured, and the system is `magic-transit`.

Cloudflare supports two types of packet captures:

> - **Sample**: Sample packets are best for debugging and providing a global picture across all data centers. Sample packets generate single, smaller files and only contain the first 160 bytes of the payload. Sampled packets are collected across all edge metals to build a PCAP file.

Reviewer 2: ```suggestion
- **Simple**: These packets are sampled, and are best for debugging and providing a global picture across all data centers. Sampled packets generate single, smaller files and only contain the first 160 bytes of the payload. Sampled packets are collected across all edge metals to build a PCAP file.
```

> - **Full**: Full packets are best for targeted data collection with a detailed view into a single data center. Full packets generate multiple large files, and they are captured within a given point-of-presence or data center and sent to either a GCP or AWS bucket specified by the user. T

Reviewer: ```suggestion
- **Full**: Full packets are best for targeted data collection with a detailed view into a single data center. Full packets generate multiple large files, and they are captured within a given data center and sent to either a GCP or AWS bucket specified by the user.
```

<details>
<summary>
  API
</summary>
 <div class="special-class" markdown="1">

> The PCAPs API needs both `system` and `type` to be specified in order to start a capture. A PCAP's `system` is the product or logical subsystem where packets are captured, and a PCAP's `type` is how the captured packets are built into a PCAP file.

Reviewer: ```suggestion
The PCAPs API needs both `system` and `type` to be specified to start a capture. A PCAP's `system` is the product or logical subsystem where packets are captured, and a PCAP's `type` is how the captured packets are built into a PCAP file.
```

> To create a sample PCAP request, send a JSON body with the required parameter listed at [Create sample PCAP request](https://api.cloudflare.com/#magic-pcap-collection-create-simple-pcap-request).

Reviewer 2: ```suggestion
To create a simple PCAP request, send a JSON body with the required parameter listed at [Create simple PCAP request](https://api.cloudflare.com/#magic-pcap-collection-create-simple-pcap-request).
```

> For full PCAP requests, refer to the required parameters listed at [Create full PCAP requests](https://api.cloudflare.com/#magic-pcap-collection-create-full-pcap-request). Note that full packet captures require two additional parameters than sample packets.

Reviewer 2: ```suggestion
For full PCAP requests, refer to the required parameters listed at [Create full PCAP requests](https://api.cloudflare.com/#magic-pcap-collection-create-full-pcap-request). Note that full packet captures require two additional parameters than simple packets.
```

> For full PCAP requests, refer to the required parameters listed at [Create full PCAP requests](https://api.cloudflare.com/#magic-pcap-collection-create-full-pcap-request). Note that full packet captures require two additional parameters than simple packets.

Reviewer: ```suggestion
For full PCAP requests, refer to the required parameters listed at [Create full PCAP requests](https://api.cloudflare.com/#magic-pcap-collection-create-full-pcap-request). Note that full packet captures require two more parameters than simple packets.
```

> The full PCAP request endpoint also contains optional fields you can use to limit the amount of packets captured. Both full and sample packet requests contain an optional `filter_v1` parameter you can use to filter packets by IPv4 Source address, for example. For a full list of the filter options, refer to the parameter lists above.

Reviewer 2: ```suggestion
The full PCAP request endpoint also contains optional fields you can use to limit the amount of packets captured. Both full and simple packet requests contain an optional `filter_v1` parameter you can use to filter packets by IPv4 Source address, for example. For a full list of the filter options, refer to the parameter lists above.
```

> Currently, you can only send one collect request per minute for sample PCAPs, and you can only have one running or pending full PCAP at a time.

Reviewer 2: ```suggestion
Currently, you can only send one collect request per minute for simple PCAPs, and you can only have one running or pending full PCAP at a time.
```

**Full PCAPs**

> `filter_v1` can be left empty to collect all packets without any filtering.

Reviewer: Passive voice, consider tweaking "Leave `filter_v1` empty to collect all packets without any filtering."

           }'
```

While the collection is in progress, the response returns the `status` field as `pending`. You must wait for the PCAP collection to complete before downloading the file. When the PCAP is ready to download, the status changes to `success`.

```bash
---
header: Full PCAP example response
---
    {
      "result": {
        "id": "7d7c88382f0b4d5daa9587aa45a1a877",
        "submitted": "2022-06-02T18:38:22.269047Z",
        "filter_v1": {},
        "time_limit": 300,
        "status": "pending",
        "type": "full",
        "system": "magic-transit",
        "packet_limit": 10000,
        "byte_limit": 100000000,
        "colo": "sfo06",
        "destination_conf": "gs://test-magic-pcaps"
      },
      "success": true,
      "errors": [],
      "messages": []
    }
```
**Simple PCAPs**

> `filter_v1` can be left empty to collect all packets without any filtering.

Reviewer: Same comment as above regarding passive voice

Author: Also fixed

           }'
```

While the collection is in progress, the response returns the `status` field as `pending`. You must wait for the PCAP collection to complete before downloading the file. When the PCAP is ready to download, the status changes to `success`.

```bash
---
header: Full PCAP example response
---
    {
      "result": {
        "id": "7d7c88382f0b4d5daa9587aa45a1a877",
        "submitted": "2022-06-02T18:38:22.269047Z",
        "filter_v1": {},
        "time_limit": 300,
        "status": "pending",
        "type": "full",
        "system": "magic-transit",
        "packet_limit": 10000,
        "byte_limit": 100000000,
        "colo": "sfo06",
        "destination_conf": "gs://test-magic-pcaps"
      },
      "success": true,
      "errors": [],
      "messages": []
    }
```

> **Sample PCAPs**

Reviewer 2: ```suggestion
**Simple PCAPs**
```

`filter_v1` can be left empty to collect all packets without any filtering.

```bash
---

> header: Sample PCAP example request

Reviewer 2: ```suggestion
header: Simple PCAP example request
```

  }'
```

The response is a JSON body that contains the details of the job running to build the packet capture. The response contains a unique identifier for the packet capture request along with the details sent in the request.

```bash
---

> header: Sample PCAP example response

Reviewer 2: ```suggestion
header: Simple PCAP example response
```

    }
```

</div>
</details>

<details>
<summary>
  Dashboard
</summary>
 <div class="special-class" markdown="1">

1. Log in to your [Cloudflare dashboard](https://dash.cloudflare.com/login) and select **Magic Transit**.

> 2. On the **Magic Transit** page next to **Packet capture**, select **Start to capture**.

Reviewer 2: ```suggestion
2. On the **Magic Transit** page next to **Packet captures**, select **Start a capture**.
```

> 3. Select the **Captures** tab and select **Start to capture**.

Reviewer 2: ```suggestion
3. Select the **Captures** tab and select **Start a capture**.
```

> 4. From **Packet Capture configuration**, choose a capture type and select **Next**.

Reviewer 2: ```suggestion
4. From **Packet captures - capture configuration**, choose a capture type and select **Next**.
```

Reviewer 2: Handle heading + subheading

5. Fill out the required fields to begin the capture and then select **Start**.

> The main **Packet Capture** page displays and displays a list of captures.

Reviewer 2: ```suggestion
The main **Packet captures** page displays and displays a list of captures.
```

> The main **Packet captures** page displays and displays a list of captures.

Reviewer: ```suggestion
The main **Packet captures** page displays a list of captures.
```

      "messages": []
    }
```
While the collection is in process, the status returns `pending` or `running`. After the PCAP is ready, the status changes to `success`.

</div>
</details>

<details>
<summary>
  Dashboard
</summary>
 <div class="special-class" markdown="1">

1. Log in to your [Cloudflare dashboard](https://dash.cloudflare.com/login) and select **Magic Transit**.

> 2. On the **Magic Transit** page next to **Packet capture**, select **Start to capture**.

Reviewer 2: ```suggestion
2. On the **Magic Transit** page next to **Packet captures**, select **Start a capture**.
```

3. From the **Packet Capture** page, select the **Captures** tab.

</div>
</details>

The capture status displays one of the following options:

- **Complete:** The capture request is done and ready for download.
- **In progress:** The capture request was captured but still processing.
- **Failure:** The capture . If this occurs, verify your ownership information.

## Download packet captures

After your request finishes processing, you can download your packet captures.

<details>
<summary>
  API
</summary>
 <div class="special-class" markdown="1">

For more information on how to process multiple saved capture files into a single output file, refer to [Wireshark's mergecap documentation](https://www.wireshark.org/docs/man-pages/mergecap.html).

**Full PCAPs**

To obtain full PCAPs, download the files from the bucket specified in `destination_conf` after the PCAP's status is `success`. You may find multiple files named `pcap_<pcap_id>.pcap` per capture as captures can occur across multiple machines.

> **Sample PCAPs**

Reviewer 2: ```suggestion
**Simple PCAPs**
```

> Once the sample PCAP collection is complete, you can download the PCAP by specifying the PCAP identifier used earlier.

Reviewer 2: ```suggestion
Once the simple PCAP collection is complete, you can download the PCAP by specifying the PCAP identifier used earlier.
```

    --output download.pcap
```

</div>
</details>

<details>
<summary>
  Dashboard
</summary>
 <div class="special-class" markdown="1">

1. Log in to your [Cloudflare dashboard](https://dash.cloudflare.com/login) and select **Magic Transit**.

> 2. On the **Magic Transit** page next to **Packet capture**, select **Start to capture**.

Reviewer 2: ```suggestion
2. On the **Magic Transit** page next to **Packet captures**, select **Start a capture**.
```

> 2. On the **Magic Transit** page next to **Packet capture**, select **Start to capture**.

Reviewer 2: ```suggestion
2. On the **Magic Transit** page next to **Packet captures**, select **Start a capture**.
```

> 3. From the **Packet Capture** page, select the **Captures** tab.

Reviewer 2: ```suggestion
3. From the **Packet captures** page, select the **Captures** tab.
```

      "messages": []
    }
```

</div>
</details>

<details>
<summary>
  Dashboard
</summary>
 <div class="special-class" markdown="1">

1. Log in to your [Cloudflare dashboard](https://dash.cloudflare.com/login) and select **Magic Transit**.

> 2. On the **Magic Transit** page next to **Packet capture**, select **Start to capture**.

Reviewer 2: ```suggestion
2. On the **Magic Transit** page next to **Packet captures**, select **Start a capture**.
```

> 3. From the **Packet Capture** page, select the **Captures** tab.

Reviewer 2: ```suggestion
3. From the **Packet captures** page, select the **Captures** tab.
```
